### DISCUSSION TITLE
integration with `folke/trouble.nvim`

### DISCUSSION THREAD

**Discussion by arctic-penguin** [neoclide/coc.nvim#4102] [Q&A]
I would like to use `folke/trouble.nvim` (https://github.com/folke/trouble.nvim) with `coc.nvim` and `coc-rust-analyzer`, because I prefer the way `trouble` shows errors and quickfixes.

I have already filed an issue here: https://github.com/folke/trouble.nvim/issues/203 but that hasn't received any attention yet.

Does anyone know if this is possible at all? Or does anyone have successfully implemented this? 

Thank you in advance!

**Comment by chemzqm** [comment]
Use `:call CocAction('diagnosticList')` to get list of quickfix item formatted diagnostics.

**Comment by arctic-penguin** [comment]
That command does not do anything for me.

I have introduced an error in a rust document. coc.nvim and coc-rust-analyzer warn me about it. When I use the command, nothing changes, and the output of `trouble.nvim` is still empty.

![screen01](https://user-images.githubusercontent.com/103587359/186885776-ad93a294-0db3-488a-9518-73b916f8b80a.png)
.

**Reply by chemzqm** [reply]
Check the result by `:echo CocAction('diagnosticList')`, you need send the result to trouble.nvim

**Reply by arctic-penguin** [reply]
Thank you!
